### PR TITLE
PHR 634 | Fetch Health Info for PHR Viewer

### PR DESCRIPTION
**Pull request by bipul-jha**

### CODE REVIEW

File: src/main/java/in/org/projecteka/hiu/dataflow/HealthInfoManager.java [hunk 1]
```diff
@@ -35,6 +45,62 @@
...
+                .flatMapMany(dataParts -> getDataEntries(limit, offset, dataParts));
+    }
+
+    private boolean isValidRequester(List<DataPartDetail> dataParts, String requesterId) {
+        return dataParts.stream().allMatch(dataPart -> dataPart.getRequester().equals(requesterId));
+    }
+
+    private Flux<PatientDataEntry> getDataEntries(int limit, int offset, List<DataPartDetail> dataParts) {
+        HashMap<String, List<HealthInfoStatus>> dataPartStatuses = new HashMap<>();
+        HashMap<String, PatientDataEntry.PatientDataEntryBuilder> dataEntries = new HashMap<>();
```
Comment (ganesanarun) [inline src/main/java/in/org/projecteka/hiu/dataflow/HealthInfoManager.java:61]: Do you really need this as a Map?

File: src/main/java/in/org/projecteka/hiu/dataflow/HealthInfoManager.java [hunk 2]
```diff
@@ -35,6 +45,62 @@
...
+        List<PatientDataEntry> partsInProcess = getDataPartsInProcess(dataPartStatuses, dataEntries);
+        return healthInformationRepository.getHealthInformation(transactionIds, limit, offset)
+                .map(healthInfo -> dataEntries.get(healthInfo.get("transaction_id").toString())
+                        .status(toStatus((String) healthInfo.get("status")))
+                        .data(healthInfo.get("data")).build())
+                .mergeWith(Flux.fromIterable(partsInProcess));
+    }
+
+    private List<PatientDataEntry> getDataPartsInProcess(HashMap<String, List<HealthInfoStatus>> dataPartStatuses,
+                                                         HashMap<String, PatientDataEntry.PatientDataEntryBuilder> dataEntries) {
```
Comment (ganesanarun) [inline src/main/java/in/org/projecteka/hiu/dataflow/HealthInfoManager.java]: I don't think you need to pass this additional map HashMap\<String, PatientDataEntry.PatientDataEntryBuilder> dataEntries. Think about it.

Comment (ganesanarun) [inline src/main/java/in/org/projecteka/hiu/dataflow/HealthInfoManager.java]: I don't think you need to pass this additional map HashMap\<String, PatientDataEntry.PatientDataEntryBuilder> dataEntries. Think about it.

File: src/main/java/in/org/projecteka/hiu/SecurityConfiguration.java [hunk 3]
```diff
@@ -53,8 +54,12 @@
 
     private static final List<Map.Entry<HttpMethod, String>> CM_PATIENT_APIS = List.of(
             Map.entry(HttpMethod.GET, "/cm/hello"),
-            Map.entry(HttpMethod.POST, APP_PATH_PATIENT_CONSENT_REQUEST));
+            Map.entry(HttpMethod.POST, APP_PATH_PATIENT_CONSENT_REQUEST),
+            Map.entry(HttpMethod.POST, HealthInfoController.API_PATH_FETCH_PATIENT_HEALTH_INFO));
 
+    private static final List<Map.Entry<HttpMethod, String>> DOCTOR_AND_PATIENT_COMMON_APIS = List.of(
+            Map.entry(HttpMethod.GET, HealthInfoController.API_PATH_GET_ATTACHMENT)
+    );
```
Comment (ganesanarun) [inline src/main/java/in/org/projecteka/hiu/SecurityConfiguration.java]: Did you change the HIU-UI to pass the token?

File: src/test/java/in/org/projecteka/hiu/dataflow/TestBuilders.java [hunk 4]
```diff
@@ -30,6 +36,15 @@
...
     }
 
+    public static HealthInformationFetchRequest.HealthInformationFetchRequestBuilder healthInformationRequest() {
+        return easyRandom.nextObject(HealthInformationFetchRequest.HealthInformationFetchRequestBuilder.class);
+    }
+
+    public static List<DataPartDetail> dataPartDetails(int size, String requester, HealthInfoStatus status) {
+        var builders = easyRandom.objects(DataPartDetail.DataPartDetailBuilder.class, size);
+        return builders.map(builder -> builder.requester(requester).status(status).build()).collect(Collectors.toList());
+    }
```
Comment (ganesanarun) [inline src/test/java/in/org/projecteka/hiu/dataflow/TestBuilders.java]: NIC: Suggestion would be to return only builders.

File: src/main/java/in/org/projecteka/hiu/SecurityConfiguration.java (hunk 3 above)
Comment (dheeraj-p) [reply src/main/java/in/org/projecteka/hiu/SecurityConfiguration.java]: Not yet. Need to change it.

File: src/test/java/in/org/projecteka/hiu/dataflow/TestBuilders.java (hunk 4 above)
Comment (dheeraj-p) [reply src/test/java/in/org/projecteka/hiu/dataflow/TestBuilders.java]: Yes sure.

File: src/main/java/in/org/projecteka/hiu/dataflow/HealthInfoManager.java (hunk 1 above)
Comment (dheeraj-p) [reply src/main/java/in/org/projecteka/hiu/dataflow/HealthInfoManager.java:61]: Yeah, we need it as a map. If you can suggest something else that would be good.

File: src/main/java/in/org/projecteka/hiu/dataflow/HealthInfoManager.java (hunk 2 above)
Comment (dheeraj-p) [reply src/main/java/in/org/projecteka/hiu/dataflow/HealthInfoManager.java]: done. Please check.

Comment (dheeraj-p) [reply src/main/java/in/org/projecteka/hiu/dataflow/HealthInfoManager.java]: done.

File: src/test/java/in/org/projecteka/hiu/dataflow/TestBuilders.java (hunk 4 above)
Comment (dheeraj-p) [reply src/test/java/in/org/projecteka/hiu/dataflow/TestBuilders.java]: done

**Review by leelanakka — APPROVED** [review]

**Pull request merged**